src/main/java/net/mcreator/element/types/Item.java

```diff
@@ -220,14 +223,105 @@ public List<StateEntry> getModels() {
 					model.stateMap.put(prop, value);
 			});
 
-			// only add this state if at least one supported property is present
-			if (!model.stateMap.isEmpty())
+			// only add this state if not duplicate and at least one supported property is present
+			if (!models.contains(model) && !model.stateMap.isEmpty()) {
+				model.listIndex = index.getAndIncrement();
 				models.add(model);
+			}
 		});
 		return models;
 	}
 
-	public static class StateEntry implements IWorkspaceDependent {
+	public StateEntry getBaseState() {
+		StateEntry baseState = new StateEntry();
+		baseState.renderType = renderType;
+		baseState.texture = texture;
+		baseState.customModelName = customModelName;
+		return baseState;
+	}
+
+	/**
+	 * Returns a tree-like states representation referencing only properties supported in the current workspace.
+	 * For use by 1.21.4+ generators.
+	 *
+	 * @return Models tree with contents matching current generator.
+	 */
+	public StateTreeNode getModelsTree() {
+		StateEntry baseState = getBaseState();
+
+		// initially set to item's default state
+		// if no model overrides are defined, this will return tree of depth 1
+		StateTreeNode root = baseState;
+		for (StateEntry state : getModels()) {
+			if (state.stateMap.isEmpty())

```

Comment:
Do we need this check?

Reply:
This is here so as to eliminate IDEA's warning about `cachedPair` possibly being null after the loop. (This statement signals that the `for` loop below will run at least once, and so will assignment of `cachedPair` at the end of it.)